service-module/src/main/java/com/gmail/pashasimonpure/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+package com.gmail.pashasimonpure.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.gmail.pashasimonpure.repository.ItemRepository;
+import com.gmail.pashasimonpure.repository.ShopRepository;
+import com.gmail.pashasimonpure.repository.model.Item;
+import com.gmail.pashasimonpure.repository.model.Shop;
+import com.gmail.pashasimonpure.service.ShopService;
+import com.gmail.pashasimonpure.service.model.ShopDTO;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class ShopServiceImpl implements ShopService {
+
+    private final ShopRepository shopRepository;
+    private final ItemRepository itemRepository;
+
+    public ShopServiceImpl(ShopRepository shopRepository, ItemRepository itemRepository) {
+        this.shopRepository = shopRepository;
+        this.itemRepository = itemRepository;
+    }
+
+    @Override
+    @Transactional
+    public ShopDTO add(ShopDTO shopDTO) {
+        Shop shop = convert(shopDTO);
+        shopRepository.persist(shop);
+        return convert(shop);
+    }
+
+    @Override
+    @Transactional
+    public ShopDTO findById(Long id) {
+        return convert(shopRepository.findById(id));
+    }
+
+    @Override
+    @Transactional
+    public List<ShopDTO> findShopsWithItemId(Long id) {
+        List<Shop> shops = shopRepository.findShopsWithItemId(id);
+        List<ShopDTO> shopsDTO = new ArrayList<>();
+
+        for (Shop shop : shops) {
+            shopsDTO.add(convert(shop));
+        }
+        return shopsDTO;
+    }
```

Comment:
If you add shops field into Item entity instead of adding items into Shop entity, you wouldn't need this complex logic, because item would have already filled shops list when you get it from DB